### DISCUSSION TITLE
Azure AKS - LoadBalancer instead of ClusterIP

### DISCUSSION THREAD

**Discussion by klemen-df** [emqx/emqx-operator#616] [Q&A] [answered]
I'm trying to use the operator to configure a single node EMQX 5. I use [terraform](https://www.terraform.io/) so don't mind the exact syntax.

I install the operator with helm and create a single instance EMQX 5 👍 

```
    "spec" = {
      "image" = "emqx/emqx:5.0.17"
      "coreTemplate" = {
        "spec" = {
          "replicas" = 1
        }
      }
      "replicantTemplate" = {
        "spec" = {
          "replicas" = 0
        }
      }
```

I configure the bootstrapConfig to include listening on the tls/8883 port :

```
        listeners.ssl.mqtts {
          bind = "0.0.0.0:8883"
          max_connections = 1024000
          ssl_options {
            keyfile = "/etc/certs/key.pem"
            certfile = "/etc/certs/cert.pem"
            versions = ["tlsv1.3", "tlsv1.2"]
          }
        }  
```

A service of type ClusterIP is created, it listens on 8883/TCP and 1883/TCP.

Is there a way to tell the operator that this should be of LoadBalancer type?

**Comment by Rory-Z** [comment, marked as answer]
You can set `.spec.listenersServiceTemplate.spec.type = LoadBalancer`, and if need add some annotations from Azure, you can set `.spec.listenersServiceTemplate.metadata.annotations`

**Comment by klemen-df** [comment]
Many thanks, this works.

Would it be also possible to setup a ClusterIP for 1883 and LoadBalancer for 8883 or do they have to be of the same type?

**Reply by Rory-Z** [reply]
It must the same type. but you can create a new service resource by yourself, and use `.spec.selector` to match EMQX pod

**Reply by klemen-df** [reply]
Should we just match :
apps.emqx.io/db-role=core
apps.emqx.io/instance=emqx
apps.emqx.io/managed-by=emqx-operator
?

**Reply by Rory-Z** [reply]
In EMQX 5, there are two types of nodes: core nodes and replicant nodes. We recommend accessing the dashboard (port 18083) through a core node and connecting MQTT clients to a replicant node (port 1883). Therefore, our CR provides `dashboardServiceTemplate` and `listenerServiceTemplate`, which you can set according to your needs.

More info, please check https://www.emqx.io/docs/en/v5.0/deploy/cluster/mria-introduction.html#mria-rlog-architecute
